### PR TITLE
[Docs] Tranlste frameworks/app/yomo.md to zh

### PR DESCRIPTION
**Pull request by zhuyaguang**
Signed-off-by: zhuyaguang <zhuyaguang1368@163.com>

### CODE REVIEW

**Comment by zhuyaguang** [comment]
@meoww-bot comment has been translated！

**Comment by alabulei1** [comment]
@zhuyaguang  Thanks for your contribution. Please check out my feedback.

File: docs/book/zh/src/frameworks/app/yomo.md [hunk 1]
```diff
@@ -1 +1,285 @@
 # YoMo
+
+[YoMo](https://yomo.run/) 是一个让开发者能够构建分布式云系统（地理分布式云系统）的编程框架。YoMo 的通信层是建立在 QUIC 协议之上，QUIC 协议带来了高速的数据传输。另外，YoMo 内置了流式 Serverless 的 “流式函数”，这可以显著提升分布式云系统的开发体验。YoMo 构建的分布式云系统提供了近场算力与终端之间的超高速通信机制， 使得它在元宇宙，VR/AR，IoT 等领域有着广泛的应用。
```
[the remark this answers is not in the document]
Comment (zhuyaguang) [reply docs/book/zh/src/frameworks/app/yomo.md:3]: 已修改

File: docs/book/zh/src/frameworks/app/yomo.md [hunk 2]
```diff
@@ -1 +1,285 @@
...
+
+> YoMo 是使用 Go 语言编写的。对于流式 Serverless，Go 语言的插件和共享库可以动态加载用户的代码，这也对开发者带来了一些限制。加上 Serverless 架构对隔离性的刚性需求，这使得 WebAssembly 成为运行用户定义函数的最佳选择。
+
+举个例子， 在 AR/VR 设备或智能工厂的 AI 推理过程中，摄像头通过  YoMo 向近场MEC（对接入边缘计算）设备中的计算节点发送实时非结构化数据。当 AI 推理完成之后，YoMo 将 AI 计算结果实时发送到终端设备。因此，托管的 AI 推理功能实现了自动化。
```
[the remark this answers is not in the document]
Comment (zhuyaguang) [reply docs/book/zh/src/frameworks/app/yomo.md:7]: 已修改

File: docs/book/zh/src/frameworks/app/yomo.md [hunk 3]
```diff
@@ -1 +1,285 @@
...
+
+然而，YoMo 面临的挑战是在边缘计算节点中整合和管理多个外部开发者开发的 handler 函数。这需要在不影响性能的情况下对这些函数进行 runtime 隔离。传统的软件容器解决方案，比如 Docker,难以胜任。因为传统的软件容器太重太慢，无法处理实时任务。
```
[the remark this answers is not in the document]
Comment (zhuyaguang) [reply docs/book/zh/src/frameworks/app/yomo.md:9]: 已修改

File: docs/book/zh/src/frameworks/app/yomo.md [hunk 4]
```diff
@@ -1 +1,285 @@
...
+
+WebAssembly 提供了一个轻量级和高性能的软件容器。 它非常适合作为 YoMo 数据处理 handler 函数的 runtime。
+
+在本文中，我们将向你展示如何创建一个基于 Tensorflow 图像分类的 Rust 函数，将其编译为 WebAssembly，然后使用 YoMo 作为流数据 handler 运行它。我们使用 [WasmEdge](https://wasmedge.org/) 作为 我们 WebAssembly runtime，因为与其他 WebAssembly runtime 相比，WasmEdge 提供了最高的性能与灵活性。是唯一能稳定支持 Tensorflow 的 WebAssembly 虚拟机。[YoMo 通过 WasmEdge 的 Go 语言 API]((https://www.secondstate.io/articles/extend-golang-app-with-webassembly-rust/)) 来管理 WasmEdge 虚拟机实例和 contained WebAssembly 字节码应用。
```
[the remark this answers is not in the document]
Comment (zhuyaguang) [reply docs/book/zh/src/frameworks/app/yomo.md:13]: the link change to https://github.com/WasmEdge/WasmEdge/blob/master/docs/book/en/src/embed/go.md ,so the original link should change ?

File: docs/book/zh/src/frameworks/app/yomo.md [hunk 5]
```diff
@@ -1 +1,285 @@
...
+
+> 源码: <https://github.com/yomorun/yomo-wasmedge-tensorflow>
```
[the remark this answers is not in the document]
Comment (zhuyaguang) [reply docs/book/zh/src/frameworks/app/yomo.md:15]: 已删除，应该是复制过来的

File: docs/book/zh/src/frameworks/app/yomo.md [hunk 6]
```diff
@@ -1 +1,285 @@
...
+```
+
+你应该把 `wasm32-wasi` 目标添加到 rust，用以将此函数编译为 WebAssembly 字节码。
+
+```bash
+rustup target add wasm32-wasi
+
+cd flow/rust_mobilenet_food
+cargo build --target wasm32-wasi --release
+# The output WASM will be target/wasm32-wasi/release/rust_mobilenet_food_lib.wasm
```
[the remark this answers is not in the document]
Comment (zhuyaguang) [reply docs/book/zh/src/frameworks/app/yomo.md:127]: OK

File: docs/book/zh/src/frameworks/app/yomo.md [hunk 7]
```diff
@@ -1 +1,285 @@
...
+
+# Copy the wasm bytecode file to the flow/ directory
```
[the remark this answers is not in the document]
Comment (zhuyaguang) [reply docs/book/zh/src/frameworks/app/yomo.md:129]: OK

File: docs/book/zh/src/frameworks/app/yomo.md [hunk 8]
```diff
@@ -1 +1,285 @@
...
+cp target/wasm32-wasi/release/rust_mobilenet_food_lib.wasm ../
+```
+
+为了释放 WasmEdge 的最佳性能，你应该将 `.wasm`  文件编译为 `.so` 文件来启用 AOT 模式。
+
+```bash
+wasmedgec rust_mobilenet_food_lib.wasm rust_mobilenet_food_lib.so
```
[the remark this answers is not in the document]
Comment (zhuyaguang) [reply docs/book/zh/src/frameworks/app/yomo.md:136]: OK

File: docs/book/zh/src/frameworks/app/yomo.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (alabulei1) [reply docs/book/zh/src/frameworks/app/yomo.md:13]: Yes. The original link also should be changed!

File: docs/book/zh/src/frameworks/app/yomo.md [hunk 9]
```diff
@@ -1 +1,285 @@
...
+
+  // 识别图像
+  res, err := bg.Execute("infer", img)
+  if err == nil {
+    fmt.Println("GO: Run bindgen -- infer:", string(res))
+  } else {
+    fmt.Println("GO: Run bindgen -- infer FAILED")
+  }
+
+  // print logs
```
[the remark this answers is not in the document]
Comment (zhuyaguang) [reply docs/book/zh/src/frameworks/app/yomo.md:202]: OK

File: docs/book/zh/src/frameworks/app/yomo.md [hunk 10]
```diff
@@ -1 +1,285 @@
...
+  hash := genSha1(img)
+  log.Printf("✅ received image-%d hash %v, img_size=%d \n", atomic.AddUint64(&counter, 1), hash, len(img))
+
+  return 0x11, nil
+}
+
+// genSha1 generate the hash value of the image
```
[the remark this answers is not in the document]
Comment (zhuyaguang) [reply docs/book/zh/src/frameworks/app/yomo.md:209]: OK

File: docs/book/zh/src/frameworks/app/yomo.md [hunk 11]
```diff
@@ -1 +1,285 @@
...
+
+然而，YoMo 面临的挑战是在边缘计算节点中整合和管理多个外部开发者开发的 handler 函数。这需要在不影响性能的情况下对这些函数进行 runtime 隔离。传统的软件容器解决方案，比如 Docker,难以胜任。因为传统的软件容器太重太慢，无法处理实时任务。
+
+WebAssembly 提供了一个轻量级和高性能的软件容器。 它非常适合作为 YoMo 数据处理 handler 函数的 runtime。
+
+在本文中，我们将向你展示如何创建一个基于 Tensorflow 图像分类的 Rust 函数，将其编译为 WebAssembly，然后使用 YoMo 作为流数据 handler 运行它。我们使用 [WasmEdge](https://wasmedge.org/) 作为 我们 WebAssembly runtime，因为与其他 WebAssembly runtime 相比，WasmEdge 提供了最高的性能与灵活性。是唯一能稳定支持 Tensorflow 的 WebAssembly 虚拟机。[YoMo 通过 WasmEdge 的 Go 语言 API]((https://www.secondstate.io/articles/extend-golang-app-with-webassembly-rust/)) 来管理 WasmEdge 虚拟机实例和 contained WebAssembly 字节码应用。
+
+> 源码: <https://github.com/yomorun/yomo-wasmedge-tensorflow>
+
+详情可查看 [YoMo 中的 WasmEdge 图像分类功能实战](https://www.youtube.com/watch?v=E0ltsn6cLIU)
```
[the remark this answers is not in the document]
Comment (zhuyaguang) [reply docs/book/zh/src/frameworks/app/yomo.md:17]: OK

File: docs/book/zh/src/frameworks/app/yomo.md [hunk 12]
```diff
@@ -1 +1,285 @@
...
+
+
+## 安装前提
+
+显然，您需要 [安装 Go 语言](https://golang.org/doc/install)，但我觉得您肯定安装好了。
+
+> 为了我们的示例代码能正常工作， Go 语言版本应不低于 1.15 。
+
+您还要安装 YoMo CLI  应用程序，其负责数据流和 handler 函数调用的编排和协调。
+
```
[the remark this answers is not in the document]
Comment (zhuyaguang) [reply docs/book/zh/src/frameworks/app/yomo.md:28]: 是不是 you 统一都翻译成你 ，那我上面的 Obviously, you will need to have [Golang installed](https://golang.org/doc/install), but I will assume you already did. 也翻译成你吧

File: docs/book/zh/src/frameworks/app/yomo.md [hunk 13]
```diff
@@ -1 +1,285 @@
...
+YoMo CLI version: v0.1.3
+```
+
+接下来，请安装 WasmEdge 和其 Tensorflow 共享库。[WasmEdge](https://wasmedge.org/) 是由云原生基金会（CNCF）托管的行业领先的 WebAssembly runtime。我们将使用它来嵌入和运行来自 YoMo 的 WebAssembly 程序。
+
+```bash
+wget -qO- https://raw.githubusercontent.com/WasmEdge/WasmEdge/master/utils/install.sh | bash -s -- -e all -p /usr/local
+```
+
+最后，因为我们的 demo WebAssembly 函数是 Rust 编写的，因此您还需要一个  [Rust 编译器](https://www.rust-lang.org/tools/install)。
```
[the remark this answers is not in the document]
Comment (zhuyaguang) [reply docs/book/zh/src/frameworks/app/yomo.md:41]: OK

File: docs/book/zh/src/frameworks/app/yomo.md [hunk 14]
```diff
@@ -1 +1,285 @@
...
+```bash
+$ go install github.com/yomorun/cli/yomo@latest
+$ yomo version
+YoMo CLI version: v0.1.3
+```
+
+接下来，请安装 WasmEdge 和其 Tensorflow 共享库。[WasmEdge](https://wasmedge.org/) 是由云原生基金会（CNCF）托管的行业领先的 WebAssembly runtime。我们将使用它来嵌入和运行来自 YoMo 的 WebAssembly 程序。
+
+```bash
+wget -qO- https://raw.githubusercontent.com/WasmEdge/WasmEdge/master/utils/install.sh | bash -s -- -e all -p /usr/local
```
[the remark this answers is not in the document]
Comment (zhuyaguang) [reply docs/book/zh/src/frameworks/app/yomo.md:38]: change to curl -sSf https://raw.githubusercontent.com/WasmEdge/WasmEdge/master/utils/install.sh | bash ;so the original link should change ?

File: docs/book/zh/src/frameworks/app/yomo.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (alabulei1) [reply docs/book/zh/src/frameworks/app/yomo.md:28]: 是的哈

**Comment by alabulei1** [comment]
LGTM!

@zhuyaguang thanks for your contribution!

@hydai Could you please help merge this PR?

**Review by hydai — APPROVED** [review]

**Pull request merged**